### PR TITLE
Upgrade go version to 1.16 for issue #412

### PR DESCRIPTION
**Pull request by lijie0123**
### What problem does this PR solve? 
Upgrade go version to 1.16  https://github.com/pingcap/tipocket/issues/412?from=from_parent_mindnote
### What is changed and how does it work?
change go mod version and base image
### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)

Code changes

 - Has Go code change
 - Has Makefile code change
 - Has Dockerfile code change

Related changes

 - Need to update the documentation

### Does this PR introduce a user-facing change?:

```release-note
NONE
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -177,7 +177,8 @@ ifeq (, $(shell which goimports))
 	set -e ;\
 	TMP_DIR=$$(mktemp -d) ;\
 	cd $$TMP_DIR ;\
-	GO111MODULE=on go get golang.org/x/tools/cmd/goimports ;\
+	GO111MODULE=on go install golang.org/x/tools/cmd/goimports@latest;\
+	go mod download golang.org/x/net\
```
Comment (mahjonp) [inline Makefile:181]: is this line needed?

File: Makefile [hunk 2]
```diff
@@ -177,7 +177,8 @@ ifeq (, $(shell which goimports))
 	set -e ;\
 	TMP_DIR=$$(mktemp -d) ;\
 	cd $$TMP_DIR ;\
-	GO111MODULE=on go get golang.org/x/tools/cmd/goimports ;\
+	GO111MODULE=on go install golang.org/x/tools/cmd/goimports@latest;\
```
Comment (mahjonp) [inline Makefile:180]:
```suggestion
	GO111MODULE=on go install golang.org/x/tools/cmd/goimports@latest ;\
```

File: Makefile (hunk 1 above)
Comment (lijie0123) [reply Makefile:181]: solved

**Review by mahjonp — APPROVED** [review]
lgtm

**Pull request merged**